azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/config/AzureComputeServiceContextModule.java

```diff
@@ -0,0 +1,127 @@
+import org.jclouds.compute.config.ComputeServiceAdapterContextModule;
+import org.jclouds.compute.domain.Hardware;
+import org.jclouds.compute.domain.NodeMetadata;
+
+import com.google.common.base.Function;
+import com.google.inject.Inject;
+import com.google.inject.TypeLiteral;
+import com.google.inject.Provides;
+
+public class AzureComputeServiceContextModule
+        extends ComputeServiceAdapterContextModule<VMDeployment, VMHardware, VMImage, Location> {
+
+   @Override
+   protected void configure() {
+      super.configure();
+
+      bind(new TypeLiteral<ComputeServiceAdapter<VMDeployment, VMHardware, VMImage, Location>>() {
+      }).to(AzureComputeServiceAdapter.class);
+      bind(new TypeLiteral<Function<VMImage, org.jclouds.compute.domain.Image>>() {
+      }).to(VMImageToImage.class);
+      bind(new TypeLiteral<Function<VMHardware, Hardware>>() {
+      }).to(VMHardwareToHardware.class);
+      bind(new TypeLiteral<Function<VMDeployment, NodeMetadata>>() {
+      }).to(DeploymentToNodeMetadata.class);
+
+      //bind(PrioritizeCredentialsFromTemplate.class).to(UseNodeCredentialsButOverrideFromTemplate.class);
+      bind(new TypeLiteral<Function<Location, org.jclouds.domain.Location>>() {
+      }).to(LocationToLocation.class);
+
+      //bind(CreateNodesInGroupThenAddToSet.class).to(GetOrCreateStorageServiceAndVirtualNetworkThenCreateNodes.class);
+
+      // to have the compute service adapter override default locations
+      install(new LocationsFromComputeServiceAdapterModule<VMDeployment, VMHardware, VMImage, Location>() {
+      });
+   }
+
+   @Singleton
+   public static class AzureComputeConstants {
+
+      @Named(OPERATION_TIMEOUT)
+      @Inject
+      private String operationTimeoutProperty;
+
+      @Named(OPERATION_POLL_INITIAL_PERIOD)
+      @Inject
+      private String operationPollInitialPeriodProperty;
+
+      @Named(OPERATION_POLL_MAX_PERIOD)
+      @Inject
+      private String operationPollMaxPeriodProperty;
+
+      @Named(TCP_RULE_FORMAT)
+      @Inject
+      private String tcpRuleFormatProperty;
+
+      @Named(TCP_RULE_REGEXP)
+      @Inject
+      private String tcpRuleRegexpProperty;
+
+      public Long operationTimeout() {
+         return Long.parseLong(operationTimeoutProperty);
+      }
+
+      public Integer operationPollInitialPeriod() {
+         return Integer.parseInt(operationPollInitialPeriodProperty);
+      }
+
+      public Integer operationPollMaxPeriod() {
+         return Integer.parseInt(operationPollMaxPeriodProperty);
+      }
+
+      public String tcpRuleFormat() {
+         return tcpRuleFormatProperty;
+      }
+
+      public String tcpRuleRegexp() {
+         return tcpRuleRegexpProperty;
+      }
+   }
+
+   @Provides
+   @Named("azureGroupId")
+   protected String getGroupId() {
+      String group =  System.getProperty("test.azurecompute-arm.groupname");
+      if (group == null)
+         group = "jCloudsGroup";
+      return group;
+   }
```

Comment:
Remove this test code from the provider code and configure the default value in the azure api metadata. The value can be overridden in the tests, if needed, just like any other property.